src/main/kotlin/net/maizegenetics/phgv2/cli/AlignAssemblies.kt

```diff
@@ -0,0 +1,320 @@
+                }
+                // We have the SAM File, call proali to align with anchorwave
+                runAnchorwaveProali(
+                    gffFile,
+                    assemblyEntry.refFasta,
+                    assemblyEntry.asmFasta,
+                    cdsFasta,
+                    assemblyEntry.refSamOutFile,
+                    asmSamFile
+                )
+            }
+        }
+
+    private fun runAnchorwaveProali(
+        gffFile: String,
+        refFasta: String,
+        asmFasta: String,
+        cdsFasta: String,
+        refSam: String,
+        asmSam: String
+    ) {
+
+        val justNameAsm = File(asmFasta).nameWithoutExtension
+        val justNameRef = File(refFasta).nameWithoutExtension
+
+        val anchorsproFile = "${outputDir}/${justNameAsm}_${justNameRef}.anchorspro"
+        // Using just the assembly name.  This facilitates programmatically creating
+        // a GVCF keyfile from the maf keyfiles.  It will be understood that the maf
+        // file name is <assemblyFastaNoExtension>.maf
+        val outputFile = "${outputDir}/${justNameAsm}.maf"
+        val builder = ProcessBuilder(
+            "conda",
+            "run",
+            "-n",
+            "phgv2-conda",
+            "anchorwave",
+            "proali",
+            "-i",
+            gffFile,
+            "-r",
+            refFasta,
+            "-as",
+            cdsFasta,
+            "-a",
+            asmSam,
+            "-ar",
+            refSam,
+            "-s",
+            asmFasta,
+            "-n",
+            anchorsproFile,
+            "-R",
+            refMaxAlignCov.toString(),
+            "-Q",
+            queryMaxAlignCov.toString(),
+            "-t",
+            threads.toString(),
+            "-o",
+            outputFile
+        )
+
+        val redirectError = "${outputDir}/proali_${justNameAsm}_outputAndError.log"
+        println("redirectError: $redirectError")
+
+        // NOTE: anchowave proali has an output file parameter, unlike minimap2, which uses
+        // command line redirection (ie >) to write the .SAM file
+        //
+        // Do not use the anchorwave output file parameter as the message redirect output
+        // for ProcessBuilder  or you'll end up with the text "AnchorWave done!" in the middle of
+        // a reference sequence, causing problems in parsing, and causing the sequence itself to be
+        // on a separate line from the data describing it.  Instead, send all program output messages
+        // to the error file.
+        // Here, we  write all program message output (ie non MAF file output) to a single file
+        // named as above.
+
+        builder.redirectOutput(File(redirectError))
+        builder.redirectError(File(redirectError))
+        println(
+            "runAnchorwaveProali proali Command for ${justNameAsm}: " + builder.command().stream()
```

Comment:
using kotlin,do we need to convert the command to a steam() here?

Reply:
Changed to joinToString